src/components/subway/subway-dashboard.js

```diff
@@ -70,14 +72,32 @@ export default class SubwayDashboard extends React.Component {
     this.setState({ limitTrains: !this.state.limitTrains });
   };
 
+  componentWillUnmount() {
+    this.clearStationTimer();
+  }
+
   updateStations(latitude, longitude) {
+    this.clearStationTimer();
+    const fetchId = this.state.newestFetchId + 1;
```

Comment:
@cdb1212 Why do you have the fetch ID now? was there a problem with just using the timer?